cmd/zfs/zfs_project.c

```diff
@@ -227,21 +226,28 @@ zfs_project_handle_dir(const char *name, zfs_project_control_t *zpc,
 	zpc->zpc_ignore_noent = B_TRUE;
 	errno = 0;
 	while (!ret && (ent = readdir(dir)) != NULL) {
+		char *fullname;
+
 		/* skip "." and ".." */
 		if (strcmp(ent->d_name, ".") == 0 ||
 		    strcmp(ent->d_name, "..") == 0)
 			continue;
 
-		if (strlen(ent->d_name) + strlen(name) >=
-		    sizeof (fullname) + 1) {
+		if (strlen(ent->d_name) + strlen(name) >= PATH_MAX + 1) {
```

Comment:
May not be more important any more, but why PATH_MAX + 1 here?  Considering that resulting string is null-terminated and "/" is inserted, I'd expect +1 to be on the left side of comparison.

Reply:
Indeed, I went ahead and made the change to move the `+1` to the other side.  Good catch.